draconic/helpers.py

```diff
@@ -275,10 +275,46 @@ def add(self, element):
             self.__approx_len__ += 1
 
         def union(self, *s):
-            if approx_len_of(self) + sum(approx_len_of(other) for other in s) > config.max_const_len:
+            if (approx_len_of(self) + sum(approx_len_of(other) for other in s)) > config.max_const_len:
                 _raise_in_context(IterableTooLong, "This set is too large")
             return SafeSet(super().union(*s))
 
+        def intersection(self, *s):
+            if approx_len_of(self) > config.max_const_len or any(approx_len_of(other) > config.max_const_len for other in s):
+                _raise_in_context(IterableTooLong, "This set is too large")
+            return SafeSet(super().intersection(*s))
+
+        def difference(self, *s):
+            if approx_len_of(self) > config.max_const_len:
+                _raise_in_context(IterableTooLong, "This set is too large")
+            return SafeSet(super().difference(*s))
+
+        def symmetric_difference(self, *s):
+            if approx_len_of(self) + sum(approx_len_of(other) for other in s) > config.max_const_len:
+                _raise_in_context(IterableTooLong, "This set is too large")
+            return SafeSet(super().symmetric_difference(*s))
+
+        def intersection_update(self, *s):
+            self.update(self.intersection(*s))
+
+        def difference_update(self, *s):
+            self.update(self.difference(*s))
+
+        def symmetric_difference_update(self, s):
+            self.update(self.symmetric_difference(s))
```

Comment:
This is not the same operation:
```py
>>> a = {1, 2, 3}
>>> a.symmetric_difference_update({2, 3, 4})
>>> a
{1, 4}

>>> a = {1, 2, 3}
>>> a.update(a.symmetric_difference({2, 3, 4}))
>>> a
{1, 2, 3, 4}
```

(it deleted my original comment for some reason)